java/memory/memory-core/src/main/java/org/apache/arrow/memory/BufferLedger.java

```diff
@@ -48,7 +48,7 @@ public class BufferLedger implements ValueWithKeyIncluded<BufferAllocator>, Refe
         "BufferLedger[%d]", 1) : null;
   private volatile long lDestructionTime = 0;
 
-  BufferLedger(final BufferAllocator allocator, final AllocationManager allocationManager) {
+  public BufferLedger(final BufferAllocator allocator, final AllocationManager allocationManager) {
```

Comment:
Ideally we'd find a way not to do this. IMO, the 'right' API might be a formal `ForeignBufferReferenceManager` implementation that is part of the core memory manager.

Reply:
Updated to take this approach.